generators/client/needle-api/needle-client-angular.js

```diff
@@ -125,11 +125,16 @@ module.exports = class extends needleClientBase {
         const iconImport = `fa${this.generator.upperFirstCamelCase(iconName)}`;
         if (!jhipsterUtils.checkRegexInFile(iconsPath, new RegExp(`\\b${iconImport}\\b`), this.generator)) {
             try {
+                let newIconContent = `\n  ${iconImport},\n  // jhipster-needle-add-icon-import`;
+                const iconNeedle = '\\r?\\n\\s*\\/\\/ jhipster-needle-add-icon-import';
+                if (!jhipsterUtils.checkRegexInFile(iconsPath, new RegExp(`,${iconNeedle}`), this.generator)) {
```

Comment:
I don’t think we should add this test.
We should add the comma to the template instead.

The only use case I can think the comma in the template doesn’t cover is a module with updated jhipster been applied to a old jhipster source.
Or I am missing something?

Reply:
Two possible use cases where trailing comma is missing:
1. update JHipster version, don't regenerate, run module which uses `addIcon`
2. user changes `.prettierrc` and sets `trailingComma: none`

First one is unlikely, second can happen, question is: should we support that or not.

If we support then code in `generator-jhipster` is more complex, if we don't support then some users can get compilation errors but these errors are very easily fixable.

So, should I remove this check to make generator simpler but risk with some minor errors in edge cases?

Reply:
If we support, we should support in a more generic way.
I suggest to add both and remove in the transform step before prettier.

Reply:
@kaidohallik what about `// prettier-ignore`?

Reply:
I think we should not put `// prettier-ignore` in the icons file. End user should get nicely formatted code.

I will remove edge cases handling from this particular place. So if someone wants to implement general trailing comma edge cases protection then this place doesn't contain redundant code.

I retain first commit so if someone wants then can investigate one possible solution.